### ISSUE TITLE
InertiaLink event callbacks not working with Vue and Laravel

### ISSUE THREAD

**Issue by shanAliKhos** [inertiajs/inertia#322]
event-callbacks are not working in my laravel 

        store() {
  
            const self = this;
            let formData = new FormData();
            formData.append("name", self.form.name); 
            formData.append("logo", self.form.logo);
            this.$inertia.post(route('admin.brand.store'), formData,{
                preserveState: true,
                preserveScroll: true,                
                onStart: () => {
                    alert('onStart')
                },
            });
        },

**Comment by reinink** [comment]
What version of Inertia.js are you on? The event callbacks were added in `0.3.0`.

**Comment by seewhy17** [comment]
I have a similar issue:
```
      methods: {
        submit() {
          this.$inertia.post(this.route('applicants.store'), this.form, {
            onError: () => alert('hmmmm!'),
            onSuccess: () => this.form.reset(),


          })
        },
```
The ```onSuccess``` call back fires at every request, even if it has validation errors, while ```onError``` won't fire at all

I am on a Laravel 8.20.1 Project (Jetstream>intertia) 
```
λ npm list --depth 0
C:\wamp64\www\freeclass
+-- @inertiajs/inertia@0.3.6
+-- @inertiajs/inertia-vue@0.2.4
+-- @inertiajs/progress@0.1.2
+-- @tailwindcss/ui@0.6.2
+-- axios@0.19.2
+-- cross-env@7.0.3
+-- laravel-jetstream@0.0.3
+-- laravel-mix@5.0.9
+-- lodash@4.17.20
+-- moment@2.29.1
+-- portal-vue@2.1.7
+-- postcss-import@12.0.1
+-- resolve-url-loader@3.1.2
+-- tailwindcss@1.9.6
+-- vue@2.6.12
+-- vue-meta@2.4.0
`-- vue-template-compiler@2.6.12
```

**Comment by claudiodekker** [comment]
@seewhy17 That's because you seem to be on `@inertiajs/inertia` `0.3.6`, whereas the `onError` callback is only available as of [`@inertiajs/inertia` 0.8.0 (see here)](https://github.com/inertiajs/inertia/releases/tag/v0.8.0). 

If you want to use `onError` (and other useful features that have been introduced since `0.3.6`, we recommend you to upgrade to the latest Inertia version. Do note that there have been a few notable breaking changes since which you might need to apply to your application, depending on what features you're using:

- https://github.com/inertiajs/inertia/releases/tag/inertia-vue%40v0.3.0
- https://github.com/inertiajs/inertia/releases/tag/v0.4.0
- https://github.com/inertiajs/inertia/releases/tag/inertia-vue%40v0.4.0
- https://github.com/inertiajs/inertia/releases/tag/v0.6.0
- https://github.com/inertiajs/inertia/releases/tag/inertia-vue%40v0.5.0
- https://github.com/inertiajs/inertia/releases/tag/v0.8.0

Additionally, since there hasn't been any reply on this issue from it's original creator, I'm assuming that they either already solved their problem or that this solves it, and as such will close this issue. If this didn't solve the issue after all, feel free to re-open it at any time.
